pkg/operator/ceph/cluster/key_rotation.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/config/keyring"
+	"github.com/rook/rook/pkg/operator/ceph/reporting"
+)
+
+// ClusterKeysToRotate defines the type of ceph daemon keys that need rotation
+type ClusterKeysToRotate struct {
+	RBDMirrorPeer bool
+}
+
+// GetKeysToRotate returns the KeysToRotate struct that defines the type of keys that need rotation.
+func (c *cluster) GetKeysToRotate() (ClusterKeysToRotate, error) {
```

Comment:
Let's do away with this helper and have each controller do its own `ShouldRotateCephxKeys()` call where needed.

We can instead make sure that `c` has current and desired ceph version info present that the method needs in subsequent calls.

Reply:
done.